react-native/src/utils/idProvider.ts

```diff
@@ -0,0 +1,82 @@
+import type { IRandomizer } from "./randomizer";
+import { type IUuidGenerator } from "./uuidGenerator";
+
+/**
+ * Abstract class defining the contract for generating tracing and session identifiers.
+ */
+export abstract class IIdProvider {
+    /** Returns a type 4 (pseudo randomly generated) UUID. */
+    public abstract uuid(): string;
+
+    /** Generates a new valid span ID (16 hex characters). */
+    public abstract spanId(): string;
+
+    /** Generates a new valid trace ID (32 hex characters). */
+    public abstract traceId(): string;
+}
+
+// ----------------------------------------------------------------------
+
+/**
+ * Concrete implementation of the IdProvider, responsible for generating
+ * unique trace, span, and session IDs.
+ */
+export class IdProvider extends IIdProvider {
```

Comment:
Review if we can directly use the otel implementation for ID generation.
https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/web-common/src/DefaultIdGenerator.ts